### PR TITLE
[CU-2705gc7] Add option to add drawables in conffeti

### PR DESCRIPTION
**Pull request by Thorusan**

### CODE REVIEW

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/ParticleSystem.kt [hunk 1]
```diff
@@ -34,6 +28,7 @@ fun CreateParticles(
     lifeTime: LifeTime = LifeTime(255f, 1f),
     emissionType: EmissionType = EmissionType.ExplodeEmission(),
     durationMillis: Int = 10000,
+    imagesList: ParticleImage? = ParticleImage.Images(emptyList())
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/ParticleSystem.kt]: `particleImage` and move to line 26. Make non-nullable.

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/ParticleConfig.kt [hunk 2]
```diff
@@ -94,14 +105,15 @@ sealed class EmissionType {
...
+    val x: Float = 0f,
+    val y: Float = 0f,
+    val velocity: Velocity,
+    val force: Force,
+    val acceleration: Acceleration,
+    val particleSize: ParticleSize,
+    val particleColor: ParticleColor,
+    val lifeTime: LifeTime,
+    val emissionType: EmissionType,
+    val image: ParticleImage?
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/ParticleConfig.kt]: `particleImage` move to line 113.

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/ParticleSystem.kt [hunk 3]
```diff
@@ -43,7 +38,16 @@ fun CreateParticles(
...
+            x,
+            y,
+            velocity,
+            force,
+            acceleration,
+            particleSize,
+            particleColor,
+            lifeTime,
+            emissionType,
+            imagesList
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/ParticleSystem.kt]: `particleImage` and move to line 46.

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/ParticleConfig.kt [hunk 4]
```diff
@@ -74,6 +75,16 @@ internal fun ParticleColor.getExactColor(): Color {
     }
 }
 
+sealed class ParticleImage {
+    data class Images(val imageBitmapList: List<ImageBitmap>) : ParticleImage()
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/ParticleConfig.kt]: rename to `images`.

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/Particle.kt [hunk 5]
```diff
@@ -16,7 +19,8 @@ internal class Particle constructor(
     var velocity: Vector2D = Vector2D(0f, 0f),
     var acceleration: Vector2D = Vector2D(0f, 0f),
     var lifetime: Float = 255f,
-    var agingFactor: Float = 20f
+    var agingFactor: Float = 20f,
+    var imageBitmap: ImageBitmap?
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/Particle.kt]: rename to `image`.

File: particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/Particle.kt [hunk 6]
```diff
@@ -46,14 +50,22 @@ internal class Particle constructor(
...
-            alpha = alpha,
-            topLeft = Offset(x, y),
-            size = Size(size, size),
-            useCenter = true
-        )
+        if (imageBitmap != null) {
+            drawScope.drawImage(
+                image = imageBitmap!!,
+                topLeft = Offset(x, y),
+                alpha = alpha
```
Comment (TimotejKovac) [inline particlesystem/src/main/java/me/nikhilchaudhari/quarks/particle/Particle.kt]: Add also size.

**Review by TimotejKovac — COMMENTED** [review]
Some minor refactoring

**Review by TimotejKovac — APPROVED** [review]
Looks good 👍

**Review by pompe84 — APPROVED** [review]
Nice

**Pull request merged**